src/amo/components/GuidesAddonCard/index.js

```diff
@@ -0,0 +1,225 @@
+          <div className="GuidesAddonCard-content">
+            <img
+              className="GuidesAddonCard-content-icon"
+              src={getAddonIconUrl(addon)}
+              alt={addon.name}
+            />
+            <div className="GuidesAddonCard-content-text">
+              <div className="GuidesAddonCard-content-header">
+                <div className="GuidesAddonCard-content-header-title">
+                  <span className="GuidesAddonCard-content-header-authors">
+                    <AddonTitle addon={addon} as="span" />
+                  </span>
+                </div>
+                {staffPick && (
+                  <div className="GuidesAddonCard-content-header-staff-pick">
+                    <Icon name="trophy" />
+                    <span>{i18n.gettext('Staff Pick')}</span>
+                  </div>
+                )}
+              </div>
+              <p className="GuidesAddonCard-content-description">
+                {this.props.addonCustomText}
+              </p>
+            </div>
+            {showInstallButton && currentVersion && (
+              <AMInstallButton
+                addon={addon}
+                currentVersion={currentVersion}
+                defaultButtonText={i18n.gettext('Add')}
+                defaultInstallSource={this.props.defaultInstallSource}
+                disabled={!compatible}
+                enable={this.props.enable}
+                hasAddonManager={this.props.hasAddonManager}
+                install={this.props.install}
+                installTheme={this.props.installTheme}
+                setCurrentStatus={this.props.setCurrentStatus}
+                status={this.props.installStatus}
+                uninstall={this.props.uninstall}
+                isAddonEnabled={this.props.isAddonEnabled}
+                puffy={false}
+              />
+            )}
+            {/* TODO: See https://github.com/mozilla/addons-frontend/issues/6901. */}
+            {showGetFirefoxButton && (
+              <Button
+                buttonType="confirm"
+                href={`https://www.mozilla.org/firefox/new/${makeQueryStringWithUTM(
+                  {
+                    utm_content: addon.guid,
+                  },
+                )}`}
+                puffy
+                className="Button--get-firefox"
+              >
+                {i18n.gettext('Only with Firefox—Get Firefox Now')}
+              </Button>
+            )}
+          </div>
+        </div>
+      </Card>
+    ) : null;
+  }
+}
+
+export const mapStateToProps = (
+  state: AppState,
+  ownProps: Props,
+): $Shape<InternalProps> => {
+  const { addon } = ownProps;
+
+  let currentVersion = null;
+  let installedAddon = {};
+
+  if (addon) {
+    installedAddon = state.installations[addon.guid];
+
+    if (addon.currentVersionId) {
+      currentVersion = getVersionById({
+        id: addon.currentVersionId,
+        state: state.versions,
+      });
+    }
+  }
+  return {
+    addon,
```

Comment:
You don't need to return this. It will already be available as `this.props.addon`.

Reply:
oops..yep, that was an oversight..thanks!

Reply:
actually we do need this for withInstallHelper it seems. I've added a comment with this.

Reply:
It is indeed needed for this HOC but if you have it in `ownProps` (or `this.props`), there is no need to return it again in `mapStateToProps`.

Reply:
well it's not working now without this

Reply:
I've explained a bit more in this comment: https://github.com/mozilla/addons-frontend/pull/6859#discussion_r233870053

Reply:
What if you invert the `connect` and `withInstallHelpers` HOCs in `compose()`?

Also, it seems like you need the `withRouter` HOC (and it should be the first HOC in the list) because `LocationType` is not `ReactRouterLocationType`.

Reply:
Thanks for the `withRouter` tip, I'll update that:)

regarding switching around compose:
```javascript
compose(
  withRouter,
  withInstallHelpers({ defaultInstallSource: INSTALL_SOURCE_GUIDES_PAGE }),
  connect(mapStateToProps),
  translate(),
)
...
```
is that what you are suggesting?

I still see this: 

`Invariant Violation: addon is required` (even if return the `addon`). 

The only way I didn't have to do this is when I checked the `addon` from the Guides page:

```javascript
 {addon && (
            <GuidesAddonCard
              addon={addon}
              addonCustomText={section.addonCustomText}
            />
          )}
```

The reason I just removed this was I thought this would cause problems/pointless setting up LoadingText in GuidesAddonCard then, right?

Reply:
That's why I suggested, at least to see what is going on. Now, it seems clear to me that `addon` can be `undefined`, which is not what the flow type suggests.

Reply:
> `addon` can be `undefined`

right that was the issue..

> Maybe you can update the code in `Guides` to comply with the Flow type:

ahh yes, that's better than what I have. cool, thanks for the help here:) I'll update..